app/src/main/java/com/cp/tasky/MainActivity.kt

```diff
@@ -30,6 +35,14 @@ class MainActivity : ComponentActivity() {
                     setUpAuthGraph(navController = navController)
                     setUpPlannerGraph(navController = navController)
                 }
+
+                if (!authViewModel.isAuthenticatedUser) {
+                    navController.navigate(AuthScreen.Route) {
+                        popUpTo(PlannerScreen.Agenda) {
+                            inclusive = true
+                        }
+                    }
+                }
```

Comment:
This is not only a side-effect that will trigger with every single recomposition as long as `isAuthenticatedUser` is false, but there's also a better way to do the initial start navigation. You can just set a conditional `startDestination`:
```
if(!authViewModel.isCheckingAuth) {
    NavHost(
        navController = navController
        startDestination = if(authViewModel.isAuthenticated) AuthGraph else AgendaGraph
    )
}
```